run-tests.py

```diff
@@ -86,6 +86,15 @@ def run_doc_tests():
             ],
             check=True,
         )
+        print("Checking for doctests.")
+        subprocess.run(
+            [
+                "pytest",
+                "--doctest-plus",
+                "pybamm",
+            ],
+            check=True,
+        )
     except subprocess.CalledProcessError as e:
```

Comment:
This subprocess call can now be made a bit cleaner in general since there would be no files to remove if Sphinx is not run. We can wrap everything in a simple try-except block and remove the `finally` block.

Reply:
Again if we do not run Sphinx, how to test for documentation builds?

Reply:
Answered above, I hope that helps